website/blog/2021-02-22-7.13.0.md

```diff
@@ -0,0 +1,255 @@
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let user, points;
+
+let score = points ?? 0;
+let name = user?.name;
+```
+
+</td>
+<td>
+
+```js
+let user, points;
+
+let score = points !== null && points !== void 0 ? points : 0;
+let name = user === null || user === void 0 ? void 0 : user.name;
+```
+
+</td>
+</tr>
+</table>
+
+However, enabling the `noDocumentAll` assumption tells Babel "I'm never using `document.all`, so you can compile without worrying about it":
+
+```json
+{
+  "targets": "firefox 70",
+  "assumptions": { "noDocumentAll": true },
+  "presets": ["@babel/preset-env"]
+}
+```
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let user, points;
+
+let score = points ?? 0;
+let name = user?.name;
+```
+
+</td>
+<td>
+
+```js
+let user, points;
+
+let score = points != null ? points : 0;
+let name = user == null ? void 0 : user.name;
+```
+
+</td>
+</tr>
+</table>
+
+Check out the full list of assumptions [over in our documentation](https://babeljs.io/assumptions), where you can individually enable or disable them to see how they affect the compiled output.
+
+### Records and Tuples support ([#12145](https://github.com/babel/babel/pull/12145))
+
+Thanks to a collaboration with [Bloomberg](https://www.techatbloomberg.com/), Babel now supports transforming the "Records and Tuples" stage 2 [proposal](https://github.com/tc39/proposal-record-tuple).
+
+The Babel plugin transforms the records and tuples syntax using the global `Record` and `Tuple` functions:
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let data = #{
+  name: "Babel",
+  ids: #[1, 2, 3]
+};
+```
+
+</td>
+<td>
+
+```js
+let data = Record({
+  name: "Babel",
+  ids: Tuple(1, 2, 3)
+});
+```
+
+</td>
+</tr>
+</table>
+
+This means that you will need to load a polyfill for those global functions, such as [`@bloomberg/record-tuple-polyfill`](https://www.npmjs.com/package/@bloomberg/record-tuple-polyfill).
+
+In order to enable this transform, you need to add `@babel/plugin-proposal-record-and-tuple` to your configuration.
+
+### New Flow features ([#12193](https://github.com/babel/babel/pull/12193), [#12234](https://github.com/babel/babel/pull/12234))
+
+Babel 7.12.0 has support for two new Flow features:
+
+- `this` type annotation in functions, that allow you to specify the type of the `this` object as if it was a parameter
+
+  ```js
+  function getPerson(this: Database, id: string): Person {
+    this instanceof Database; // true
+  }
+  ```
+
+- `enum` declarations with unknown members
+
+  ```js
+  enum PetKind {
+    Dog,
+    Cat,
+    Snake,
+    ...
+  }
+  ```
+
+### TypeScript 4.2 support ([#12628](https://github.com/babel/babel/pull/12628))
+
+[TypeScript 4.2](https://devblogs.microsoft.com/typescript/announcing-typescript-4-2-rc/) is currently RC, and it supports some new syntax features such as abstract constructor signatures.
```

Comment:
Should we clarify what "RC" means?

Reply:
We can just delete it, tomorrow it will be released as stable anyway.
```suggestion
[TypeScript 4.2](https://devblogs.microsoft.com/typescript/announcing-typescript-4-2-rc/) supports some new syntax features such as abstract constructor signatures.
```